File: python/send.py
import os
import requests
import re

def sc_send(sendkey, title, desp='', options=None):
    if options is None:
        options = {}
    # 判断 sendkey 是否以 'sctp' 开头，并提取数字构造 URL
    if sendkey.startswith('sctp'):
        match = re.match(r'sctp(\d+)t', sendkey)
        if match:
            num = match.group(1)
            url = f'https://{num}.push.ft07.com/send/{sendkey}.send'
        else:
            raise ValueError('Invalid sendkey format for sctp')
    else:
        url = f'https://sctapi.ftqq.com/{sendkey}.send'
    params = {
        'title': title,
        'desp': desp,
        **options
    }
    headers = {
        'Content-Type': 'application/json;charset=utf-8'
    }
    response = requests.post(url, json=params, headers=headers)
    result = response.json()
    return result


data = {}
with open(os.path.join(os.path.dirname(__file__), '..', '.env'), 'r') as f:
    for line in f:
        key, value = line.strip().split('=')
        data[key] = value
key = data['SENDKEY']

ret = sc_send(key, '主人服务器宕机了 via python', '第一行\n\n第二行')
print(ret)

File: nodejs/send.js
const https = require('https');
const querystring = require('querystring');
const fs = require('fs');
const path = require('path');

const data = require('dotenv').parse(fs.readFileSync( path.join(__dirname, '../.env') ));
const key = data.SENDKEY;

(async () => {
    const ret = await sc_send('主人服务器宕机了', "第一行\n\n第二行", key);
    console.log(ret);
})();

async function sc_send(text, desp = '', key = '[SENDKEY]') {
    const postData = querystring.stringify({ text, desp });
    // 根据 sendkey 是否以 'sctp' 开头，选择不同的 API URL
    const url = String(key).startsWith('sctp') 
        ? `https://${key}.push.ft07.com/send`
        : `https://sctapi.ftqq.com/${key}.send`;
  
    const response = await fetch(url, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded',
        'Content-Length': Buffer.byteLength(postData)
      },
      body: postData
    });
  
    const data = await response.text();
    return data;
  }